lib/email_tracking_retriever.py

```diff
@@ -80,7 +81,11 @@ def get_trackings(self) -> Dict[str, Tracking]:
             if success:
               for new_tracking in new_trackings:
                 trackings[new_tracking.tracking_number] = new_tracking
+            elif attempt < MAX_ATTEMPTS - 1:
```

Comment:
we could probably do something like 

```
if success:
  <add to map as we already do>
  break
elif attempt >= MAX_ATTEMPTS:
  tqdm.write("Failed to find tracking number...")
  # normal failure stuff as before
```

i think that might be cleaner

Reply:
or even factor this out into a separate method, then we could replace the break with a return and we wouldn't need an else statement for the failure (would just have it after the loop) but that's probably overkill

Reply:
Putting the clauses in order as success, retry, failure makes the most sense to me.  Success, failure, retry seems out of order and less clear.

Reply:
well in my case, the retry is implied -- it means there's only two cases

Reply:
Done.